core/execution/update_market_state_test.go

```diff
@@ -0,0 +1,124 @@
+package execution_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"code.vegaprotocol.io/vega/core/execution"
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/num"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVerifyUpdateMarketState(t *testing.T) {
+	now := time.Now()
+	exec := getEngine(t, now)
+	pubKey := &types.SignerPubKey{
+		PubKey: &types.PubKey{
+			Key: "0xDEADBEEF",
+		},
+	}
+	mkt := newMarket("MarketID", pubKey)
+	err := exec.engine.SubmitMarket(context.Background(), mkt, "", time.Now())
+	require.NoError(t, err)
+
+	config := &types.MarketStateUpdateConfiguration{
+		MarketID:   "wrong",
+		UpdateType: types.MarketStateUpdateTypeTerminate,
+	}
+
+	require.Equal(t, execution.ErrMarketDoesNotExist, exec.engine.VerifyUpdateMarketState(config))
+
+	config.MarketID = mkt.ID
+	require.Equal(t, fmt.Errorf("missing settlement price for governance initiated futures market termination"), exec.engine.VerifyUpdateMarketState(config))
+}
+
+func TestTerminateMarketViaGovernance(t *testing.T) {
+	now := time.Now()
+	exec := getEngine(t, now)
+	pubKey := &types.SignerPubKey{
+		PubKey: &types.PubKey{
+			Key: "0xDEADBEEF",
+		},
+	}
+	mkt := newMarket("MarketID", pubKey)
+	err := exec.engine.SubmitMarket(context.Background(), mkt, "", time.Now())
+	require.NoError(t, err)
+
+	exec.engine.StartOpeningAuction(context.Background(), mkt.ID)
+
+	config := &types.MarketStateUpdateConfiguration{
+		MarketID:        mkt.ID,
+		UpdateType:      types.MarketStateUpdateTypeTerminate,
+		SettlementPrice: num.NewUint(100),
+	}
+	require.NoError(t, exec.engine.UpdateMarketState(context.Background(), config))
+	state, err := exec.engine.GetMarketState(mkt.ID)
+	require.NoError(t, err)
+	require.Equal(t, types.MarketStateClosed, state)
+}
+
+func TestSuspendMarketViaGovernance(t *testing.T) {
+	now := time.Now()
+	exec := getEngine(t, now)
+	pubKey := &types.SignerPubKey{
+		PubKey: &types.PubKey{
+			Key: "0xDEADBEEF",
+		},
+	}
+	mkt := newMarket("MarketID", pubKey)
+	err := exec.engine.SubmitMarket(context.Background(), mkt, "", time.Now())
+	require.NoError(t, err)
+
+	exec.engine.StartOpeningAuction(context.Background(), mkt.ID)
+
+	// during opening auction
+	state, err := exec.engine.GetMarketData(mkt.ID)
+	require.NoError(t, err)
+	require.Equal(t, types.MarketStateActive, state.MarketState)
+	require.Equal(t, types.MarketTradingModeContinuous, state.MarketTradingMode)
+
+	config := &types.MarketStateUpdateConfiguration{
+		MarketID:        mkt.ID,
+		UpdateType:      types.MarketStateUpdateTypeSuspend,
+		SettlementPrice: num.NewUint(100),
+	}
+	require.NoError(t, exec.engine.UpdateMarketState(context.Background(), config))
+
+	// after governance suspension
+	state, err = exec.engine.GetMarketData(mkt.ID)
+	require.NoError(t, err)
+	require.Equal(t, types.MarketStateSuspendedViaGovernance, state.MarketState)
+	require.Equal(t, types.MarketTradingModeSuspendedViaGovernance, state.MarketTradingMode)
+
```

Comment:
Can we throw an `OnTick()` somewhere in here between the suspension and resume to make sure it stays in the correct state for a block before resuming?